src/core/model_repository_manager.cc

```diff
@@ -86,13 +90,30 @@ BuildBackendConfigMap(
       graphdef_config->per_process_gpu_memory_fraction = tf_gpu_memory_fraction;
     }
 
+#ifdef TRTIS_ENABLE_GPU
+    int device_cnt = 0;
+    cudaError_t cuerr = cudaGetDeviceCount(&device_cnt);
+    if (cuerr == cudaErrorNoDevice) {
+      device_cnt = 0;
+    } else if (cuerr != cudaSuccess) {
+      LOG_ERROR << "unable to get number of CUDA devices while building "
+                   "BackendConfigMap: "
+                << cudaGetErrorString(cuerr);
+    }
```

Comment:
set device_cnt = 0 since not sure what cudaGetDeviceCount did to it in error case